packages/block-editor/src/components/inner-blocks/warning-max-depth-exceeded.native.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { TouchableWithoutFeedback, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import Warning from '../warning';
+import BlockFallbackWebVersion from '../block-fallback-web-version';
+import { store as blockEditorStore } from '../../store';
+
+const WarningMaxDepthExceeded = ( { clientId } ) => {
+	const [ showDetails, setShowDetails ] = useState( false );
+
+	const { isSelected, innerBlocks } = useSelect(
+		( select ) => {
+			const { getBlock, isBlockSelected } = select( blockEditorStore );
+			return {
+				innerBlocks: getBlock( clientId )?.innerBlocks || [],
+				isSelected: isBlockSelected( clientId ),
+			};
+		},
+		[ clientId ]
+	);
+	const { replaceBlocks } = useDispatch( blockEditorStore );
+
+	const onUngroup = () => {
+		if ( ! innerBlocks.length ) {
+			return;
+		}
+
+		replaceBlocks( clientId, innerBlocks );
+	};
+
+	/* translators: Warning related to having blocks deeply nested. %d: The deepest nesting level. */
+	const descriptionFormat = __(
+		'Blocks with inner blocks that are nested deeper than %d levels might not be rendered properly in the native version of the editor. For this reason, we recommend flattening the content by ungrouping the block or editing the block using the web editor.'
+	);
+	const description = sprintf( descriptionFormat, MAX_NESTING_DEPTH );
+
+	return (
+		<TouchableWithoutFeedback
+			disabled={ ! isSelected }
+			accessibilityLabel={ __( 'Warning message' ) }
+			accessibilityRole={ 'button' }
+			accessibilityHint={ __( 'Tap here to show more details.' ) }
+			onPress={ () => setShowDetails( true ) }
+		>
+			<View>
+				<Warning
+					message={ __(
+						'Block cannot be rendered because it is deeply nested. Tap here for more details.'
+					) }
+				/>
+				<BlockFallbackWebVersion
+					clientId={ clientId }
+					showSheet={ showDetails }
+					onCloseSheet={ () => setShowDetails( false ) }
+					title={ __( 'Block deeply nested' ) }
```

Comment:
This suggestion is pretty much the same sentence, but for whatever reason this potential alternative sounds better to me. Possibly pure preference. Please feel free to ignore the suggestion. 
```suggestion
					title={ __( 'Deeply nested block' ) }
```